subgraphs/platypus-finance/src/common/getters.ts

```diff
@@ -0,0 +1,341 @@
+import { BigInt, log } from "@graphprotocol/graph-ts";
+import { Address, BigDecimal, ethereum } from "@graphprotocol/graph-ts";
+import {
+  Token,
+  DexAmmProtocol,
+  LiquidityPool,
+  UsageMetricsDailySnapshot,
+  UsageMetricsHourlySnapshot,
+  FinancialsDailySnapshot,
+  LiquidityPoolDailySnapshot,
+  LiquidityPoolHourlySnapshot,
+  _LiquidityPoolParamsHelper,
+  _Asset,
+  RewardToken,
+  LiquidityPoolFee,
+} from "../../generated/schema";
+import { Pool as PoolTemplate } from "../../generated/templates";
+import { fetchTokenSymbol, fetchTokenName, fetchTokenDecimals } from "./tokens";
+import {
+  Network,
+  PROTOCOL_ADMIN,
+  ProtocolType,
+  BIGDECIMAL_ZERO,
+  SECONDS_PER_DAY,
+  SECONDS_PER_HOUR,
+  poolDetail,
+  RewardTokenType,
+  LiquidityPoolFeeType,
+} from "../common/constants";
+import { exponentToBigDecimal } from "./utils/numbers";
+import { getUsdPrice } from "../prices";
+
+let altPoolsInit = false;
+
+export function initAltPoolTemplates(event: ethereum.Event): void {
+  // Start watching the LiquidityPools
+  // Note: I have no idea what happens if I create a dynamic datasource
```

Comment:
Did you figure out this one?

Reply:
its runs all mappings twice

Reply:
can you add that as a comment so people are aware?

Reply:
I'll actually remove the note, there are no duplicate datasources. Maybe I can add this method to the development docs somewhere.